tools/system_libs.py

```diff
@@ -889,7 +870,7 @@ def can_build(self):
     return super(crt1, self).can_build() and shared.Settings.WASM_BACKEND
 
 
-class libc_extras(MuslInternalLibrary):
+class libc_extras(NoBCLibrary, MuslInternalLibrary):
```

Comment:
For some libraries yes.   The advantage of the .bc library is: everything is in single file.  The disadvantage is: The linker can't skip over it and will always reference it (including any static constructors therein).    The whole point of libc_extras I believe is so that its not part of the main libc.bc file and can be selectively included.     It only contains the single file: `extras_fastcomp.c`.   

With this change we starting including it on the link line every time, which has the effect that it cannot be ignored.. by making it into a `.a` archive the linker process can ignore it when its not needed.

Reply:
Sorry I might be mistaken, but does fastcomp use `.a` files? I thought it only uses `.bc`...

Reply:
Fastcomp and handle .a files, but the code that handles it is not very nice.  It has to extract the archive and into a temp directory and inspect each object file.  This is (I believe) why it uses .bc files for things like libc, a an optimization.   I mean I can't think of any other reason to make libc into into big .bc file.  Perhaps @kripken can remember the history here.

Reply:
I think historically we just had a single `.bc` file for every library. That was easiest as we just let `llvm-link` do the linking.

Later we made some libraries be `.a`, where it mattered, like libc++ which has global ctors, and we wrote python code to extract the archive and pick the members to link etc. (as we couldn't use `llvm-link` there).